### PR TITLE
doc: replace lifo/fifo with LIFO/FIFO

### PR DESCRIPTION
**Pull request by nashif**
Replace all occurances of lifo/fifo with LIFO/FIFO to be consistent.

### CODE REVIEW

File: doc/reference/kernel/data_passing/fifos.rst [hunk 1]
```diff
@@ -32,26 +32,26 @@ reserved space requirements for data items if they are added with
...
 
-A data item may be **added** to a fifo by a thread or an ISR.
+A data item may be **added** to a FIFO by a thread or an ISR.
 The item is given directly to a waiting thread, if one exists;
-otherwise the item is added to the fifo's queue.
+otherwise the item is added to the FIFO's queue.
 There is no limit to the number of items that may be queued.
 
-A data item may be **removed** from a fifo by a thread. If the fifo's queue
+A data item may be **removed** from a FIF by a thread. If the FIFO's queue
```
Comment (galak) [inline doc/reference/kernel/data_passing/fifos.rst:40]: lossed an 'O' on this one.

File: doc/reference/kernel/data_passing/lifos.rst [hunk 2]
```diff
@@ -76,15 +76,15 @@ The following code has the same effect as the code segment above.
...
 
-The following code builds on the example above, and uses the lifo
+The following code builds on the example above, and uses the LIFO
 to send data to one or more consumer threads.
 
 .. code-block:: c
 
     struct data_item_t {
-        void *lifo_reserved;   /* 1st word reserved for use by lifo */
+        void *LIFO_reserved;   /* 1st word reserved for use by LIFO */
```
Comment (galak) [inline doc/reference/kernel/data_passing/lifos.rst:87]: should leave variable name case alone.

File: include/kernel.h [hunk 3]
```diff
@@ -4861,7 +4861,7 @@ enum _poll_types_bits {
 	/* semaphore availability */
 	_POLL_TYPE_SEM_AVAILABLE,
 
-	/* queue/fifo/lifo data availability */
+	/* Queue/FIFO/LIFO data availability */
```
Comment (galak) [inline include/kernel.h:4864]: do you mean to change queue?

**Review by galak — CHANGES_REQUESTED** [review]
In general looks good, two issues need fixing.  One question about Queue cap change.

File: include/kernel.h (hunk 3 above)
Comment (nashif) [reply include/kernel.h:4864]: Yes

**Review by galak — APPROVED** [review]

File: include/kernel.h [hunk 4]
```diff
@@ -4880,10 +4880,10 @@ enum _poll_states_bits {
 	/* semaphore is available */
 	_POLL_STATE_SEM_AVAILABLE,
 
-	/* data is available to read on queue/fifo/lifo */
+	/* data is available to read on Queue/FIFO/LIFO */
```
Comment (nzmichaelh) [inline include/kernel.h:4883]: nit: here queue should start with a lower case q.

**Review by nzmichaelh — APPROVED** [review]

**Pull request merged**